addons/test_mail_sms/tests/test_sms_composer.py

```diff
@@ -120,6 +120,25 @@ def test_composer_comment_field_w_template(self):
 
         self.assertSMSNotification([{'partner': self.test_record.customer_id, 'number': self.test_record.mobile_nbr}], 'Dear %s this is an SMS.' % self.test_record.display_name, messages)
 
+    def test_composer_comment_invalid_field(self):
+        """ Test the Send Message in SMS Composer when a Model does not contain a number field name """
+        test_record = self.env['mail.test.sms.partner'].create({
+            'name': 'Test',
+            'customer_id': self.partner_1.id,
+        })
+        sms_composer = self.env['sms.composer'].create({
+            'body': self._test_body,
+            'number_field_name': 'phone_nbr',
+            'recipient_single_number_itf': self.random_numbers_san[0],
+            'res_id': test_record.id,
+            'res_model': 'mail.test.sms.partner'
+        })
+
+        self.assertNotIn(','.join(test_record._fields), 'phone_nbr')
```

Comment:
No need to join. Unlike js you can check that a string is in a list directly, and it's actually more efficient (and more intuitive)

Reply:
It doesn't work because the ``_fields`` returns a dict and not a list :/